### PR TITLE
Remove hardware-model based and default DeviceNetworkConfig from nim

### PR DESCRIPTION
**Pull request by eriknordmark**
This results in less churn when nim needs to test.

### CODE REVIEW

File: pkg/pillar/cmd/nim/nim.go [hunk 1]
```diff
@@ -34,10 +33,7 @@ import (
 )
 
 const (
-	agentName   = "nim"
-	tmpDirname  = "/var/tmp/zededa"
-	DNCDirname  = tmpDirname + "/DeviceNetworkConfig"
-	DPCOverride = tmpDirname + "/DevicePortConfig/override.json"
```
Comment (gkodali-zededa) [inline pkg/pillar/cmd/nim/nim.go]: So, do we not support DPC configuration override after this?

File: pkg/pillar/cmd/nim/nim.go [hunk 2]
```diff
@@ -182,40 +152,18 @@ func Run() {
...
-	subDeviceNetworkConfig.ModifyHandler = devicenetwork.HandleDNCModify
-	subDeviceNetworkConfig.DeleteHandler = devicenetwork.HandleDNCDelete
-	nimCtx.SubDeviceNetworkConfig = subDeviceNetworkConfig
-	subDeviceNetworkConfig.Activate()
+	dnc := &nimCtx.DeviceNetworkContext
+	devicenetwork.IngestPortConfigList(dnc)
 
 	// We get DevicePortConfig from three sources in this priority:
 	// 1. zedagent publishing NetworkPortConfig
 	// 2. override file in /var/tmp/zededa/NetworkPortConfig/*.json
```
Comment (gkodali-zededa) [inline pkg/pillar/cmd/nim/nim.go]: Looks like we removed code that handles the override.json case. Should we remote this comment also?

Comment (eriknordmark) [reply pkg/pillar/cmd/nim/nim.go]: Same as above; has been in /config/DevicePortConfig and /var/tmp/zededa/DevicePortConfig.

I see that some comments refer to NetworkPortConfig, and those should be changed to refer to DevicePortConfig.

File: pkg/pillar/cmd/nim/nim.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/nim/nim.go]: We, but the code isn't using the specific filename any more, nor do we use DeviceNetworkConfig for override. We use a DevicePortConfig for override.

**Pull request merged**